### ISSUE TITLE
Move `_adjustTotalSupplyCheckpoints` inside of `createCheckpoint`

### ISSUE THREAD

**Issue by adamdossa** [PolymathNetwork/polymath-core#520]
[in the dev-3.0.0 branch]

Currently we call `_adjustTotalSupplyCheckpoints` every time there is an operation that mints or burns tokens. However, in practice this will always be a no-op due to:
```
        if ((_checkpoints.length > 0) && (_checkpoints[_checkpoints.length - 1].checkpointId == _currentCheckpointId)) {
            return;
        }
```
in TokenLib.sol, unless a new checkpoint has been created since the last mint / burn operation.

It would be more efficient (in terms of gas consumption) to do this inside of the `createCheckpoint` instead.

Following the change all of the tests should still run to success, specifically `c_checkpoints.js`.

**Comment by dev1644** [comment]
I am working on it.

**Comment by maxsam4** [comment]
Fixed by #523
